src/ConnectedMode/Migration/IUnintrusiveBindingController.cs

```diff
@@ -19,19 +19,46 @@
  */
 
 using System.ComponentModel.Composition;
+using System.Threading;
+using SonarLint.VisualStudio.ConnectedMode.Binding;
 using SonarLint.VisualStudio.Core.Binding;
 
+using Task = System.Threading.Tasks.Task;
+
 namespace SonarLint.VisualStudio.ConnectedMode.Migration
 {
     internal interface IUnintrusiveBindingController
     {
-        void Bind(BoundSonarQubeProject project);
+        Task BindAsync(BoundSonarQubeProject project, CancellationToken token);
     }
 
     [Export(typeof(IUnintrusiveBindingController))]
-    [PartCreationPolicy(CreationPolicy.Shared)]
-    public class UnintrusiveBindingController : IUnintrusiveBindingController
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    internal class UnintrusiveBindingController : IUnintrusiveBindingController
     {
-        public void Bind(BoundSonarQubeProject project) { }
+        private readonly IBindingProcessFactory bindingProcessFactory;
+
+        [ImportingConstructor]
+        public UnintrusiveBindingController(IBindingProcessFactory bindingProcessFactory)
+        {
+            this.bindingProcessFactory = bindingProcessFactory;
+        }
+
+        public async Task BindAsync(BoundSonarQubeProject project, CancellationToken token)
```

Comment:
For now this doesn't report any progress. The way I see it we could either directly do it in here or do it in the BindingProcess. Either way this is a different ticket problem